src/webview/pipeline/app/element/selectwidget.ts

```diff
@@ -0,0 +1,148 @@
+      const input = new EditItem('URL', new InputWidget('Please provide Name/URL', null, this.initialValue), 'input-resource', 'inner-editItem');
+      event.appendChild(input.getElement());
+    } else if (event.lastElementChild.firstElementChild.id.trim() === 'input-resource') {
+      event.lastElementChild.remove();
+    }
+    if (event.parentNode.firstElementChild.textContent === TknResourceType.GitResource || event.parentNode.firstElementChild.textContent === TknResourceType.ImageResource) {
+      if (select.value.trim() !== 'Create Pipeline Resource') {
+        createResourceJson(event.firstElementChild.id, select.value, this.initialValue);
+      }
+    }
+    if (event.parentNode.firstElementChild.textContent === TknResourceType.Workspaces || event.parentNode.parentNode.firstElementChild.textContent === TknResourceType.Workspaces) {
+      this.clickEvent(event);
+    }
+    disableButton(document.getElementsByTagName('input'));
+    this.createWorkspaceElement(event, select);
+  }
+
+  createWorkspaceElement(event: Node & ParentNode, select: HTMLSelectElement): void {
+    try {
+      const sectionId = `${select.value}-Workspaces`;
+      const editId = 'Workspaces-Edit';
+      const optionId = select[select.selectedIndex].id;
+      if (this.trigger[select.value]) {
+        if (event.lastElementChild.firstElementChild.id.trim() === editId) event.lastElementChild.remove();
+        const workspacesType = new SelectWidget(sectionId, this.trigger, null, this.initialValue).workspacesResource(this.trigger[select.value], select.value);
+        const workspacesOp = new EditItem(VolumeTypes[select.value], workspacesType, editId, 'inner-editItem');
+        event.appendChild(workspacesOp.getElement());
+        selectText(event.querySelectorAll(`[id^=${sectionId}]`), `Select a ${VolumeTypes[select.value]}`, true, 'select-workspace-option');
+        disableSelection(document.getElementsByTagName('select'));
+      } else if (event.lastElementChild.firstElementChild.id.trim() === editId) {
+        event.lastElementChild.remove();
+        disableSelection(document.getElementsByTagName('select'));
+      }
+      if (optionId) {
+        const selectedItem = event.querySelectorAll('[id^=items-section-workspace-new-item]');
+        const buttonItem = event.querySelectorAll('.elementButtons');
+        if (optionId === 'select-workspace-option') {
+          selectedItem.forEach(element => element.remove());
+          buttonItem.forEach(element => element.remove());
+        }
+        if (selectedItem.length) {
+          selectedItem.forEach(element => element.remove());
+          buttonItem.forEach(element => element.remove());
+          createItem(event, optionId, select.value, this.initialValue, this.trigger);
+        } else {
+          createItem(event, optionId, select.value, this.initialValue, this.trigger);
+        }
+      }
+    } catch (err) {
+      // ignores
```

Comment:
What error can be catch there?

Reply:
In this try-catch I check the workspace content ex: config map, secret, and PVC, if it is not workspace content then it throws error so I am ignoring that error here.